src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -130,6 +134,11 @@ private static void AddFieldCodeActions(ArrayBuilder<CodeAction> result, Semanti
         {
             if (state.TypeToGenerateIn.TypeKind != TypeKind.Interface)
             {
+                if (!state.CanGeneratePropertyOrField())
+                {
+                    return;
+                }
```

Comment:
should this be checked prior to even calling AddFieldCodeActions and AddPropertyCodeActions ?

Reply:
if this does stay in this helper, i woudl say this should be the absolute first check in teh method.  after all, if someone calls AddField, the first thing we should do is check CanGenField and bail if not.